image/rhel/konflux.Dockerfile

```diff
@@ -19,10 +19,8 @@ COPY --from=final-base / "$FINAL_STAGE_PATH"
 COPY ./scripts/konflux/subscription-manager/* /tmp/.konflux/
 RUN /tmp/.konflux/subscription-manager-bro.sh register "$GO_BUILDER_STAGE_PATH" "$FINAL_STAGE_PATH"
 
-RUN subscription-manager repos --enable codeready-builder-for-rhel-8-x86_64-rpms
-
 # Install packages for the Go builder stage.
-RUN dnf -y --installroot="$GO_BUILDER_STAGE_PATH" install --allowerasing make automake gcc gcc-c++ coreutils binutils diffutils gflags snappy-devel zlib-devel bzip2-devel lz4-devel cmake libzstd-devel zstd jq
+RUN dnf -y --installroot="$GO_BUILDER_STAGE_PATH" install --allowerasing make automake gcc gcc-c++ coreutils binutils diffutils zlib-devel bzip2-devel lz4-devel cmake libzstd-devel zstd jq
```

Comment:
From this line, only `libzstd-devel` comes from RHEL repos. All others seem to be available in UBI repos. Is there anything that still uses `libzstd`?

Reply:
We have a dependency on `github.com/DataDog/zstd` but it needs a go tag `external_libzstd` per its [documentation](https://github.com/DataDog/zstd?tab=readme-ov-file#building-against-an-external-libzstd) in order to use external `libzstd` instead of a vendored one. I'm not able to find `external_libzstd` anywhere in our repos.

Reply:
Also https://redhat-internal.slack.com/archives/C033SKHG9CZ/p1715334926631639

Reply:
`github.com/klauspost/compress` dependency provides `zstd` implementation in Go.

Reply:
Knocked out a few more things, but I wasn't yet able to get rid of the rhel image entirely.

Reply:
Let's try this: https://github.com/stackrox/stackrox/pull/11090

Reply:
Ah, I had most of that here in an iteration and it didn't work.  it was the latter `go-builder` I had messed up.  Let me rebase and incorporate your changes.